### PR TITLE
Fix GetInternalTesTaskBlobUrlWithoutSasToken()

### LINKED ISSUE #509: DefaultStorageAccessProvider.GetInternalTesTaskBlobUrlWithoutSasToken() is broken
**Describe the bug**
In `DefaultStorageAccessProvider`, `GetInternalTesTaskBlobUrlWithoutSasToken` and `GetInternalTesTaskBlobUrlWithoutSasToken` return two different containers when `internal_path_prefix` is set on the `TesTask`.

**Steps to Reproduce**
Run a workflow with cromwell:86 and observe that the task-related log blobs are on a different container than the rest of the task-related log blobs (the logs are in the `tes-internal` container under a path that starts with the name of the execution container).

**Expected behavior**
All task-related blobs that by default are placed together in the `tes-internal` container are placed together in the path provided by the task's `internal_path_prefix`.

### PR DESCRIPTION
**Pull request by BMurri**
fixes #509

### CODE REVIEW

File: src/TesApi.Web/Storage/DefaultStorageAccessProvider.cs [hunk 1]
```diff
@@ -114,52 +115,68 @@ public override async Task<string> MapLocalPathToSasUrlAsync(string path, Cancel
...
+
+                if (pathSegments.IsContainer || getContainerSas)
+                {
+                    var policy = new SharedAccessBlobPolicy()
                     {
-                        var policy = new SharedAccessBlobPolicy() { Permissions = SharedAccessBlobPermissions.Read, SharedAccessExpiryTime = DateTime.Now.Add((sasTokenDuration ?? TimeSpan.Zero) + SasTokenDuration) };
-                        resultPathSegments.SasToken = new CloudBlob(resultPathSegments.ToUri(), new StorageCredentials(storageAccountInfo.Name, accountKey)).GetSharedAccessSignature(policy, null, null, SharedAccessProtocol.HttpsOnly, null);
-                    }
+                        Permissions = SharedAccessBlobPermissions.Add | SharedAccessBlobPermissions.Create | SharedAccessBlobPermissions.List | SharedAccessBlobPermissions.Read | SharedAccessBlobPermissions.Write,
+                        SharedAccessExpiryTime = DateTime.Now.Add((sasTokenDuration ?? TimeSpan.Zero) + SasTokenDuration)
```
Comment (MattMcL4475) [inline src/TesApi.Web/Storage/DefaultStorageAccessProvider.cs:151]: is this math correct?  `sasTokenDuration + SasTokenDuration`

File: src/TesApi.Web/Storage/DefaultStorageAccessProvider.cs [hunk 2]
```diff
@@ -114,52 +115,68 @@ public override async Task<string> MapLocalPathToSasUrlAsync(string path, Cancel
...
-                    return resultPathSegments.ToUriString();
+                    var containerUri = new StorageAccountUrlSegments(storageAccountInfo.BlobEndpoint, pathSegments.ContainerName).ToUri();
+                    resultPathSegments.SasToken = new CloudBlobContainer(containerUri, new StorageCredentials(storageAccountInfo.Name, accountKey)).GetSharedAccessSignature(policy, null, SharedAccessProtocol.HttpsOnly, null);
                 }
-                catch (Exception ex)
+                else
                 {
-                    Logger.LogError(ex, $"Could not get the key of storage account '{pathSegments.AccountName}'. Make sure that the TES app service has Contributor access to it.");
-                    return null;
+                    var policy = new SharedAccessBlobPolicy() { Permissions = SharedAccessBlobPermissions.Read, SharedAccessExpiryTime = DateTime.Now.Add((sasTokenDuration ?? TimeSpan.Zero) + SasTokenDuration) };
```
Comment (MattMcL4475) [inline src/TesApi.Web/Storage/DefaultStorageAccessProvider.cs:159]: `DateTime.UtcNow` ?

**Review by MattMcL4475 — APPROVED** [review]

File: src/TesApi.Web/Storage/DefaultStorageAccessProvider.cs (hunk 1 above)
Comment (BMurri) [reply src/TesApi.Web/Storage/DefaultStorageAccessProvider.cs]: Old error, but I'll fix it

File: src/TesApi.Web/Storage/DefaultStorageAccessProvider.cs (hunk 2 above)
Comment (BMurri) [reply src/TesApi.Web/Storage/DefaultStorageAccessProvider.cs]: ditto

**Pull request merged**